app/controllers/clients_controller.rb

```diff
@@ -0,0 +1,46 @@
+class ClientsController < ApplicationController
+  def index
+    @client = Client.all
+  end
+  
+  def new
+    @client = Client.new
+  end
+  
+  def create
+    @client = Client.new(client_params)
+      if @client.save
+        flash[:success] = "Client created successfully!"
```

Comment:
no need to set the flash here. just pass the flash message as argument to `redirect_to` method.

`redirect_to clients_path, success: "Client created successfully!" `